### PR TITLE
Add files

### PR DESCRIPTION
**Pull request by shirgoldbird**

### CODE REVIEW

File: interim_results.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+audio_queue = asyncio.Queue()
+
+def callback(input_data, frame_count, time_info, status_flag):
+    audio_queue.put_nowait(input_data)
+    return (input_data, pyaudio.paContinue)
+
+async def run(key):
+    clock_cursor = 0.
+    audio_cursor = 0.
+    transcript_cursor = 0.
```
Comment (nikolawhallon) [inline interim_results.py]: is measuring latency necessary for this demo?

File: interim_results.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                                beepy.beep(sound=1)
+                                transcript = ''
+                            latest_final_result_processed = True
+                    else:
+                        if transcript_cursor - last_word_end > SILENCE_INTERVAL:
+                            if len(transcript) > 0:
+                                print(f'Clock: {clock_cursor:07.3f}, Audio: {audio_cursor:07.3f}, Transcript: {transcript_cursor:07.3f}; {transcript}')
+                                beepy.beep(sound=1)
+                                transcript = ''
+                            latest_final_result_processed = True
```
Comment (nikolawhallon) [inline interim_results.py]: this algorithm only gets the case where the trigger happens from an interim result following a final result - it doesn't demonstrate the case where the trigger ought to happen in a final result (there are at least 2 edge cases there)

File: interim_results.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            asyncio.ensure_future(sender(ws)),
+            asyncio.ensure_future(receiver(ws))
+        ])
+
+def parse_args():
+    """ Parses the command-line arguments.
+    """
+    parser = argparse.ArgumentParser(description='Submits data to the real-time streaming endpoint.')
+    parser.add_argument('-k', '--key', required=True, help='YOUR_DEEPGRAM_API_KEY (authorization)')
+    return parser.parse_args()
```
Comment (nikolawhallon) [inline interim_results.py]: should `SILENCE_INTERVAL` also be configurable?

File: speech_final.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    clock_cursor = 0.
+    audio_cursor = 0.
+    transcript_cursor = 0.
+
+    extra_headers={
+        'Authorization': 'Token {}'.format('key')
+    }
+    try:
+        async with websockets.connect('wss://api.deepgram.com/v1/listen'):
+            print('success')
```
Comment (nikolawhallon) [inline speech_final.py]: is this incomplete? it doesn't seem to do anything

**Review by nikolawhallon — CHANGES_REQUESTED** [review]
a few things to think about

File: interim_results.py (hunk 1 above)
Comment (shirgoldbird) [reply interim_results.py]: Ah, no. I can pull it out.

File: interim_results.py (hunk 3 above)
Comment (shirgoldbird) [reply interim_results.py]: oh yeah, I can set it up so it defaults to 2.0 but can be configured via CLI

File: speech_final.py (hunk 4 above)
Comment (shirgoldbird) [reply speech_final.py]: OOPS. I screwed this up purposely for the TWIP demo. Lemme fix

File: interim_results.py (hunk 2 above)
Comment (nikolawhallon) [reply interim_results.py]: I think this (simplified algorithm) is more correct:

```
import json

def find_index(messages):
    last_word_end = 0.0
    latest_final_result_processed = False
    
    index = 0
    for message in messages:
        message = json.loads(message)
        
        transcript_cursor = message['start'] + message['duration']

        # if there are any words in the message
        if len(message['words']) > 0:
            # if there is more than 1 word in the message, look to see if there is a gap of 2 seconds between any of the words
            if len(message['words']) > 1:
                previous_word_end = message['words'][0]['end']
                should_break = False
                for word in message['words']:
                    current_word_start = word['start']
                    if current_word_start - previous_word_end >= 2.0:
                        should_break = True
                        break
                    previous_word_end = word['end']
                if should_break:
                    break

            # if the last word in a previous message is 2 or more seconds older than the first word in this message
            current_word_begin = message['words'][0]['start']
            if current_word_begin - last_word_end >= 2.0 and last_word_end != 0:
                break
            last_word_end = message['words'][-1]['end']
        # if there were no words in this message, check if the the last word in a previous message is 2 or more seconds older than the timestamp at the end of this message
        elif transcript_cursor - last_word_end >= 2.0 and last_word_end != 0:
                break
                        
        index += 1

    return index
```

Comment (shirgoldbird) [reply interim_results.py]: Updated PR with this algorithm

File: interim_results.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                rate = RATE,
+                input = True,
+                frames_per_buffer = CHUNK,
+                stream_callback = callback
+            )
+
+            stream.start_stream()
+
+            while stream.is_active():
+                await asyncio.sleep(0.1)
```
Comment (nikolawhallon) [inline interim_results.py:44]: is this in my original script? I wonder what the function of this is

File: interim_results.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                raise
+
+        async def receiver(ws):
+            transcript = ''
+            last_word_end = 0.0
+            should_beep = False
+            is_silence = False
+
+            async for message in ws:
+                print(is_silence)
```
Comment (nikolawhallon) [inline interim_results.py]: is this necessary?

**Review by nikolawhallon — APPROVED** [review]
9.5/10 there's a little something for everyone

**Pull request merged**